### PR TITLE
fix(sync): ping func should not try to read response body

### PR DESCRIPTION
**Pull request by peusebiu**
closes: #1703

**What type of PR is this?**
bug

**Which issue does this PR fix**:
Sync spams with json parsing errors in log messages, everytime it tries to ping docker, because it read the response body.

**What does this PR do / Why do we need it**:
improve ping func()

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/sync/httpclient/client.go [hunk 1]
```diff
@@ -70,13 +70,34 @@ func (httpClient *Client) SetConfig(config Config) error {
...
 	}
 
-	return true
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusUnauthorized {
+		return true
+	}
+
+	httpClient.log.Error().Str("url", pingURL.String()).Int("statusCode", resp.StatusCode).
```
[the remark this answers is not in the document]
Comment (peusebiu) [reply pkg/extensions/sync/httpclient/client.go:97]: ok, sure

File: pkg/extensions/sync/httpclient/client.go [hunk 2]
```diff
@@ -70,13 +71,42 @@ func (httpClient *Client) SetConfig(config Config) error {
...
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		httpClient.log.Error().Err(err).Str("url", pingURL.String()).
+			Msg("failed to read body while pinging registry")
+
+		return false
+	}
+
+	if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusUnauthorized {
```
Comment (andaaron) [inline pkg/extensions/sync/httpclient/client.go:102]: I would check the status before the body is read.

**Review by andaaron — APPROVED** [review]

**Review by andaaron — DISMISSED** [review]
LGTM

**Comment by rchincha** [comment]
Pls rebase and push again.

**Review by andaaron — APPROVED** [review]

**Comment by rchincha** [comment]
https://go.dev/src/net/http/client.go

```

// Do sends an HTTP request and returns an HTTP response, following

// policy (such as redirects, cookies, auth) as configured on the

// client.

//

// An error is returned if caused by client policy (such as

// CheckRedirect), or failure to speak HTTP (such as a network

// connectivity problem). A non-2xx status code doesn't cause an

// error.

//

// If the returned error is nil, the Response will contain a non-nil

// Body which the user is expected to close. If the Body is not both

// read to EOF and closed, the Client's underlying RoundTripper

// (typically Transport) may not be able to re-use a persistent TCP

// connection to the server for a subsequent "keep-alive" request.

//

// The request Body, if non-nil, will be closed by the underlying

// Transport, even on errors.

//

// On error, any Response can be ignored. A non-nil Response with a

// non-nil error only occurs when CheckRedirect fails, and even then

// the returned Response.Body is already closed.

//

// Generally Get, Post, or PostForm will be used instead of Do.

//

// If the server replies with a redirect, the Client first uses the

// CheckRedirect function to determine whether the redirect should be

// followed. If permitted, a 301, 302, or 303 redirect causes

// subsequent requests to use HTTP method GET

// (or HEAD if the original request was HEAD), with no body.

// A 307 or 308 redirect preserves the original HTTP method and body,

// provided that the Request.GetBody function is defined.

// The NewRequest function automatically sets GetBody for common

// standard library body types.

```

**Pull request merged**